glib/src/subclass/object.rs

```diff
@@ -133,6 +133,75 @@ unsafe extern "C" fn dispose<T: ObjectImpl>(obj: *mut gobject_ffi::GObject) {
     }
 }
 
+#[derive(Debug)]
+pub struct MissingPropertyHandler<'a>(&'a str);
+
+impl<'a> MissingPropertyHandler<'a> {
+    fn name(&self) -> &'a str {
+        self.0
+    }
+}
+impl<'a> From<&'a ParamSpec> for MissingPropertyHandler<'a> {
+    fn from(p: &'a ParamSpec) -> Self {
+        Self(p.name())
+    }
+}
+
+impl<'a> std::fmt::Display for MissingPropertyHandler<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "missing handler for property {}", self.name())
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// Trait containing only the property related functions of `ObjectImpl`.
+/// Implemented by the `Props` macro.
+/// When implementing `ObjectImpl` you may want to delegate the function calls to this trait.
+/// Some functions have a different return type from `ObjectImpl` to handle possible errors caused by missing
+/// properties.
+pub trait DerivedObjectProperties: ObjectSubclass {
+    // rustdoc-stripper-ignore-next
+    /// Properties installed for this type.
+    fn derived_properties() -> &'static [ParamSpec] {
+        &[]
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Similar to [`ObjectImpl`](trait.ObjectImpl.html) but returns a `Result`.
+    fn derived_set_property<'a>(
+        &self,
+        _obj: &Self::Type,
+        _id: usize,
```

Comment:
We could probably just remove the `id` arg here and on `derived_property` since they won't be used